src/converters/app-converter.ts

```diff
@@ -0,0 +1,388 @@
+                    // TODO: we could do better here. Currently the stack depends on all assets, but really
+                    // only individual resources should depend on individual assets
+                    dependsOn: this.stackDependsOn(dependencies),
+                });
+                this.constructs.set(n.construct, this.stackResource);
+                continue;
+            }
+
+            const parent = this.constructs.get(n.construct.parent!)!;
+            if (n.resource && n.logicalId) {
+                const cfn = n.resource;
+                debug(`Processing node with template: ${JSON.stringify(cfn)}`);
+                debug(`Creating resource for ${n.logicalId}`);
+                const props = this.processIntrinsics(cfn.Properties);
+                const options = this.processOptions(cfn, parent);
+
+                const mapped = this.mapResource(n.logicalId, cfn.Type, props, options);
+                const resource = pulumi.Resource.isInstance(mapped) ? mapped : mapped.resource;
+                const attributes = pulumi.Resource.isInstance(mapped) ? undefined : mapped.attributes;
+                this.resources.set(n.logicalId, { resource, attributes, resourceType: cfn.Type });
+                this.constructs.set(n.construct, resource);
+
+                debug(`Done creating resource for ${n.logicalId}`);
+                // TODO: process template conditions
+                // for (const [conditionId, condition] of Object.entries(cfn.Conditions || {})) {
+                //     // Do something with the condition
+                // }
+            } else {
+                const r = new CdkConstruct(`${this.host.name}/${n.construct.path}`, n.construct.type, {
+                    parent,
+                });
+                this.constructs.set(n.construct, r);
+            }
+        }
+
+        // Register the outputs as outputs of the component resource.
+        for (const [outputId, args] of Object.entries(this.stack.outputs ?? {})) {
+            this.host.registerOutput(outputId, this.processIntrinsics(args.Value));
+        }
+
+        for (let i = dependencyGraphNodes.length - 1; i >= 0; i--) {
+            const n = dependencyGraphNodes[i];
+            if (!n.resource) {
+                (<CdkConstruct>this.constructs.get(n.construct)!).done();
+            }
+        }
+    }
+
+    private stackDependsOn(dependencies: Set<ArtifactConverter>): pulumi.Resource[] {
+        const dependsOn: pulumi.Resource[] = [];
+        for (const d of dependencies) {
+            if (d instanceof FileAssetManifestConverter) {
+                this.resources.set(d.id, { resource: d.file, resourceType: d.resourceType });
+                dependsOn.push(d.file);
+            }
+            // TODO: handle docker images
+        }
+        return dependsOn;
+    }
+
+    private mapParameter(logicalId: string, typeName: string, defaultValue: any | undefined) {
+        // TODO: support arbitrary parameters?
```

Comment:
I'm not sure, doesn't CDK recommend against regular parameters?

AWS specific parameter types are a neat way to do some validation though.

Reply:
Yeah typically you would not use them. I think this is a very easy deferment. We could even remove this completely and say we don't support it until we get a lot of customer feedback.